File: Data Structures and Algorithms (Java & Python)/08 Hands On Activity 1/BirthMonth.py
set1 = set()
set2 = set()
set3 = set()

for i in range(3):
    month1 = input("Enter birth month " + str(i+1) + ": ")
    set1.add(month1)

for i in range(3):
    month2 = input("Enter birth month " + str(i+1) + ": ")
    set2.add(month2)
    
print("Group 1: " , set1)
print("Group 2: " , set2)

month3 = input("Enter your birth month: ")
set3.add(month3)

print("Union: " , str(set1 | set2))
print("Intersection: " , str(set1 & set2))
print("Difference: " , str(set1 - set2))

if (set3.issubset(set1 | set2)):
	print("You have the same birth month with one of your classmates")
else:
	print("You don't have the same birth month with one of your classmates")

File: Data Structures and Algorithms (Java & Python)/08 Hands On Activity 1/BirthMonth.java
import java.util.Scanner;
import java.util.Set;
import java.util.HashSet;
import java.util.Collections;

public class BirthMonth
{
	public static void main(String[] args)
	{
		Set group1 = new HashSet();
		Set group2 = new HashSet();
		Set self = new HashSet();
		Scanner sc = new Scanner(System.in);
		
		for(int i=0; i<3; i++) {
			System.out.print("Enter birth month " + (i+1) + ": ");
			String scan = sc.nextLine();
			group1.add(scan);
		}
		for(int i=0; i<3; i++) {
			System.out.print("Enter birth month " + (i+1) + ": ");
			String scan2 = sc.nextLine();
			group2.add(scan2);
		}

		System.out.println("Group 1: " + group1);
		System.out.println("Group 2: " + group2);

		System.out.print("Enter your birth month: ");
		String scan3 = sc.nextLine();

		self.add(scan3);

		Set union = new HashSet(group1);
		Set inter = new HashSet(group1);
		Set diff = new HashSet(group1);

		union.addAll(group2);
		inter.retainAll(group2);
		diff.removeAll(group2);

		System.out.println("Union: " + union);
		System.out.println("Intersection: " + inter);
		System.out.println("Difference: " + diff);

		if (union.containsAll(self))
		{
			System.out.println("You have the same birth month with one of your classmates");
		}
		else
		{
			System.out.println("You don't have the same birth month with one of your classmates");
		}

	}
}